### DISCUSSION TITLE
Max auto-tick performance is less than expected.

### DISCUSSION THREAD

**Discussion by Emil-DV** [logisim-evolution/logisim-evolution#1406] [General]
I've been building a set of complex circuits and the overall performance is coming no where near the max auto tick rate of 2048Khz.

At first I just thought this was par for the course for a large circuit but then I ran a circuit consisting of a single clock element.  The best tick rate I've seen is about 3.5khz w/ mouse off window and ~2khz with the mouse over the window.

Has anyone know of a configuration with a tick rate in the 100Khz+ range?

**Comment by R3dst0ne** [comment]
This problem already got discussed in #582.
A possible enhancement is proposed in #786.
